object-manipulation-2/src/pick.js

```diff
@@ -0,0 +1,12 @@
+/* exported pick */
+function pick(source, keys) {
+  const object = {};
+  for (let i = 0; i < keys.length; i++) {
+    for (const key in source) {
+      if (keys[i] === key && source[keys[i]] !== undefined) {
+        object[keys[i]] = source[keys[i]];
+      }
+    }
+  }
+  return object;
+}
```

Comment:
We actually only need one loop here! Look at the snippet below
```
function pick(source, keys) {
  const picked = {};
  for (let i = 0; i < keys.length; i++) {
    const key = keys[i];
    if (typeof source[key] !== 'undefined') {
      picked[key] = source[key];
    }
  }
  return picked;
}
```

Reply:
Ahh okay cool, I am finding it's easy for me to think there should more loops than are needed. Thanks!